app/src/main/java/fi/bitrite/android/ws/ui/util/HostFilterManager.java

```diff
@@ -0,0 +1,152 @@
+  public static final String RECENT_ACTIVITY_FILTER_KEY = "recentActivityFilter";
+  public static final String CURRENTLY_AVAILABLE_FILTER_KEY = "currentlyAvailableFilter";
+
+  private final HashMap<String, HostFilter> mFilters = new HashMap();
+
+  @Inject
+  HostFilterManager() {
+      mFilters.put(RECENT_ACTIVITY_FILTER_KEY, new RecentActivityFilter());
+      mFilters.put(CURRENTLY_AVAILABLE_FILTER_KEY, new CurrentlyAvailableFilter());
+  }
+
+  public boolean filterHost(SimpleUser host) {
+      for (HostFilter filter : mFilters.values()) {
+          if (!filter.filterHost(host)) {
+              return false;
+          }
+      }
+      return true;
+  }
+
+  public boolean isAnyFilterActive() {
+      for (HostFilter filter : mFilters.values()) {
+          if (filter.isActive()) {
+              return true;
+          }
+      }
+      return false;
+  }
+
+  public class InvalidFilterException extends RuntimeException { }
+
+  public void activateFilter(String filterKey) {
+      try {
+          mFilters.get(filterKey).activateFilter();
+      } catch (NullPointerException e) {
+          throw new InvalidFilterException();
+      }
+  }
+
+  public void deactivateFilter(String filterKey) {
+      try {
+          mFilters.get(filterKey).deactivateFilter();
+      } catch (NullPointerException e) {
+          throw new InvalidFilterException();
+      }
+  }
+
+  public boolean isFilterActive(String filterKey) {
+      try {
+          return mFilters.get(filterKey).isActive();
+      } catch (NullPointerException e) {
+          throw new InvalidFilterException();
+      }
+  }
+
+  public void updateFilterValue(String filterKey, int value) {
+      try {
+          mFilters.get(filterKey).updateFilterValue(value);
+      } catch (NullPointerException e) {
+          throw new InvalidFilterException();
+      } // UnsupportedOperationException is passed through
+  }
+
+  public int getFilterValue(String filterKey) {
+      try {
+          return mFilters.get(filterKey).getFilterValue();
+      } catch (NullPointerException e) {
+          throw new InvalidFilterException();
+      } // UnsupportedOperationException is passed through
+  }
+
+  abstract class HostFilter {
+      protected boolean isActive = false;
+
+      void activateFilter() {
+          isActive = true;
+      }
+      void deactivateFilter() {
+          isActive = false;
+      }
+      boolean isActive() {
+          return isActive;
+      }
+      void updateFilterValue(int value) {
+        throw new UnsupportedOperationException();
+      }
+      int getFilterValue() {
+        throw new UnsupportedOperationException();
+      }
+
+      boolean filterHost(SimpleUser host) {
+          if (!isActive) {
+              return true;
+          }
+          return filterHostInternal(host);
+      }
+      abstract boolean filterHostInternal(SimpleUser host);
+  }
+
+  // A filter that returns true iff the host is currently available.
```

Comment:
Typo.

Reply:
I thought that the contraction iff for "if and only if" was pretty standard? I'm happy to make the change if you find it confusing.

Reply:
Never heard about it, my education about formal logic was held in German. But cool, now I can understand https://xkcd.com/1033/.